sdk/storage/storage-blob/test/blobclient.spec.ts

```diff
@@ -117,9 +117,11 @@ describe("BlobClient", () => {
     await blobSnapshotClient.delete();
     await blobClient.delete();
 
-    const result2 = await containerClient.listBlobFlatSegment(undefined, {
-      include: ["snapshots"]
-    });
+    const result2 = (await (await containerClient
+      .listBlobsFlat({
```

Comment:
Double await looks bad. I'd rather do two separated await calls with local variables.

Also if we are not verifying the segment, It looks better to gather all the items in an array then verify them.

```typescript
    const all = [];
    for await (const blob of containerClient.listBlobsFlat()) {
      all.push(blob);
    }
    assert.equal(all.length, 0);
```

Reply:
We should verify the segment, right ?

Reply:
Depends on the tests.  The verification of this test is check whether all snapshots are deleted so we shouldn't verify segment.  For tests that return pages and check the responses we need to verify segments.

---
In reply to: [296358960](https://github.com/Azure/azure-sdk-for-js/pull/3991#discussion_r296358960) [](ancestors = 296358960)

Reply:
but one `await` should look good enough.

---
In reply to: [296363010](https://github.com/Azure/azure-sdk-for-js/pull/3991#discussion_r296363010) [](ancestors = 296363010,296358960)

Reply:
Fixed double await